packages/kbn-handlebars/index.ts

```diff
@@ -312,9 +319,27 @@ class ElasticHandlebarsVisitor extends Handlebars.Visitor {
       this.ast = Handlebars.parse(this.template!);
     }
 
-    this.accept(this.ast);
+    let main: Handlebars.HelperDelegate = () => {
+      const prog = this.generateProgramFunction(this.ast!);
+      return prog(this.context, this.runtimeOptions);
+    };
 
-    return this.output.join('');
+    // Run any decorators that might exist on the root
+    main = this.processDecorators(this.ast, main);
```

Comment:
__question:__ any reason we need to split `main` declaration in two parts (`let` + new assignment) instead of something like this?
```ts
    const main = this.processDecorators(this.ast, () => {
      const prog = this.generateProgramFunction(this.ast!);
      return prog(this.context, this.runtimeOptions);
    });
```

Reply:
Good point. I'll combine them 👍

Reply:
I guess it was to better show what it was the `main` function that by default was returned from `processDecorators` - or potentially overwritten with a custom `main` function. By inlining it, it's no longer as clear. But maybe that doesn't matter?

Reply:
> But maybe that doesn't matter?

I'd say so, but feel free to keep it as you did initially if you think there is a value.

Reply:
I ended up adding a bunch more comments to this function to better articulate what was going on